### PR TITLE
Fine-tune iOS's scroll start feel

### LINKED ISSUE #16529: Gestures lag at edge of screen
When dragging on a list view from the edge of the screen on iOS there is a noticeable delay at the beginning of gesture recognition.

See sample code in https://gist.github.com/tvolkert/05616985d86f90350ff51077c744e00d

lag

### PR DESCRIPTION
**Pull request by xster**
Fixes #16529

This makes sure on iOS that:

1. Slow creeping scroll moves still don't gradually drift from finger
1. Hesitant drag starts don't jump by too many pixels initially
1. Determined drag starts has closer to the inertia feeling on iOS
1. Strong flings don't lose the responsive feel
1. Scroll position is still somewhat 'sticky' when becoming stationary

### CODE REVIEW

**Comment by Hixie** [comment]
How does this code end up only affecting iOS?

**Comment by xster** [comment]
Only the iOS's scroll behaviour physics sets a `motionStartDistanceThreshold` on the ScrollDragController. Otherwise, all drag motions transparently falls through to `applyUserOffset`.

**Comment by Hixie** [comment]
One thing we might want to do is change our gesture recognizers to not do the jump, by having them start at zero offset when they "win". (Presumably we'd make that configurable, and the question here is just what the default configuration should be for those created by a GestureDetector.)

cc @HansMuller who was interested in that particular idea.

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 1]
```diff
@@ -314,19 +321,32 @@ class ScrollDragController implements Drag {
...
       // Not moving can't break threshold.
-      return false;
+      return 0.0;
     } else {
       if (_offsetSinceLastStop == null) {
-        // Already in motion. Allow transparent offset transmission.
-        return true;
+        // Already in motion or no threshold behavior configured.
+        // Allow transparent offset transmission.
+        return offset;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: Can you add some comments in this method that talk about how by default, Android falls down one path and iOS the other, and point to the class that sets things up that way? Thanks.

**Comment by Hixie** [comment]
LGTM

**Comment by xster** [comment]
I think it's sensible to make it tiered (partially gated here and partially gated in the gesture detector) and also add something similar to the gesture detector. 
We'd indeed want them to be configurable. Otherwise, winning at zero will make deliberate flings feel sluggish. 

In the long run, I still want to model iOS scrolls specifically to #13171 (it's just really hard to reverse engineer it) but that wouldn't apply to general gesture detectors so we still need 2 paths I think. 

But having it on gestures will solve a similar problem with things like back swipe which also 'jumps' in Flutter and doesn't natively.

**Pull request merged**